sound/soc/sof/topology.c

```diff
@@ -1306,6 +1306,8 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 	swidget->num_source_pins = 1;
 	swidget->num_sink_pins = 1;
 	swidget->num_pin_bindings = 0;
+	ida_init(&swidget->source_queue_ida);
+	ida_init(&swidget->sink_queue_ida);
```

Comment:
do we need to call ida_destroy in sof_widget_unload ?

Reply:
The ida source_queue_ida and ida sink_queue_ida are not pointers, they will be freed along with snd_sof_widget free.

Reply:
Please check the implementation of ida. The pointer is included by ida so that it can fit for any size of data. If there is no pointer, how to fit for any size ?

Reply:
checked several other drivers that use ida, and you are rignt. thank you.